lib/matplotlib/pyplot.py

```diff
@@ -1038,115 +1038,125 @@ def subplot(*args, **kwargs):
 
-    Keyword arguments:
-
-      *nrows* : int
-        Number of rows of the subplot grid.  Defaults to 1.
-
-      *ncols* : int
-        Number of columns of the subplot grid.  Defaults to 1.
-
-      *sharex* : string or bool
-        If *True*, the X axis will be shared amongst all subplots.  If
-        *True* and you have multiple rows, the x tick labels on all but
-        the last row of plots will have visible set to *False*
-        If a string must be one of "row", "col", "all", or "none".
-        "all" has the same effect as *True*, "none" has the same effect
-        as *False*.
-        If "row", each subplot row will share a X axis.
-        If "col", each subplot column will share a X axis and the x tick
-        labels on all but the last row will have visible set to *False*.
-
-      *sharey* : string or bool
-        If *True*, the Y axis will be shared amongst all subplots. If
-        *True* and you have multiple columns, the y tick labels on all but
-        the first column of plots will have visible set to *False*
-        If a string must be one of "row", "col", "all", or "none".
-        "all" has the same effect as *True*, "none" has the same effect
-        as *False*.
-        If "row", each subplot row will share a Y axis and the y tick
-        labels on all but the first column will have visible set to *False*.
-        If "col", each subplot column will share a Y axis.
-
-      *squeeze* : bool
-        If *True*, extra dimensions are squeezed out from the
-        returned axis object:
-
-        - if only one subplot is constructed (nrows=ncols=1), the
-          resulting single Axis object is returned as a scalar.
-
-        - for Nx1 or 1xN subplots, the returned object is a 1-d numpy
-          object array of Axis objects are returned as numpy 1-d
-          arrays.
-
-        - for NxM subplots with N>1 and M>1 are returned as a 2d
-          array.
-
-        If *False*, no squeezing at all is done: the returned axis
-        object is always a 2-d array containing Axis instances, even if it
-        ends up being 1x1.
-
-      *subplot_kw* : dict
+    Parameters
+    ----------
+    nrows : int, optional, default: 1
+        Number of rows of the subplot grid.
+
+    ncols : int, optional, default: 1
+        Number of columns of the subplot grid.
+
+    sharex : bool or string, optional, default: False
+        - If True, the X axis will be shared amongst all subplots. Note that
+          if nrows > 1 then the x tick labels won't be displayed on any of
+          plots but the ones on the bottom row.
+        - If False, no axis will be shared amongst subplots.
+        - If a string must be one of "row", "col", "all", or "none".
+            - "all" has the same effect as True.
+            - "none" has the same effect as False.
+            - If "row", each subplot row will share a X axis.
+            - If "col", each subplot column will share a X axis. Note that if
```

Comment:
This comment (which is also present for "if True...") could possibly be factored out to the end of the list: "if the X axis is shared across columns (`sharex=True` or `sharex="col"`), then etc." (and similarly for sharey).